src/puppetlabs/trapperkeeper/services/webrouting/webrouting_service_core.clj

```diff
@@ -123,6 +123,14 @@
         add-servlet-handler (:add-servlet-handler webserver-service)]
     (add-servlet-handler servlet path opts)))
 
+(schema/defn ^:always-validate add-websocket-handler!
+  [context webserver-service
+   svc :- (schema/protocol tk-services/Service)
+   handlers options :- RouteOption]
```

Comment:
@richardc I think that is what @fpringvaldsen is suggesting, yes.  I'd recommend going ahead and pushing that change up.

@fpringvaldsen if what @richardc pushes up isn't quite what you were thinking, maybe you could go ahead and add one more commit to take it to the finish line and then merge it in?  That way we won't have to wait through another 24-hour cycle to get this one wrapped up.